Tests/TrackerRadarKitPerformanceTests/NextTrackerDataSetPerformanceTests.swift

```diff
@@ -0,0 +1,166 @@
+    var tdsRefFileName: String?
+    var tdsRefURL: String?
+    
+    let numberOfRuns = 10
+    var numberOfIterationsPerRun: Int = 1 
+    
+    func nextURL(filename: String, fileURL: String) -> URL {
+        let baseURL = URL(string: fileURL)!
+        return baseURL.appendingPathComponent(filename)
+    }
+
+    override func setUp() async throws {
+        try await super.setUp()
+        
+        try loadParameters()
+        
+        // Prepare TDS Under Test file
+        let utTdsUrl = nextURL(filename: tdsUtFileName, fileURL: tdsUtURL)
+        print("TDS under Test: \(utTdsUrl.absoluteString)")
+        
+        let (data, _) = try await URLSession.shared.data(from: utTdsUrl)
+        utTDS = try JSONDecoder().decode(TrackerData.self, from: data)
+        
+        // Prepare reference TDS file
+        if let refFileName = tdsRefFileName, let refURL = tdsRefURL {
+            let refTdsUrl = nextURL(filename: refFileName, fileURL: refURL)
+            print("Reference TDS: \(refTdsUrl.absoluteString)")
+            
+            let (refData, _) = try await URLSession.shared.data(from: refTdsUrl)
+            refTDS = try JSONDecoder().decode(TrackerData.self, from: refData)
+        }
+        
+        print("TDS files prepared")
+    }
+    
+    func testPerformanceOfNext_iOSTDS() throws {
+        let utAverage = try runPerformanceTest(tds: utTDS, name: "UT TDS")
+        
+        if let refTDS = refTDS {
+            let refAverage = try runPerformanceTest(tds: refTDS, name: "Reference TDS")
+            
+            let percentDifference = (utAverage - refAverage) / refAverage
+            print("Percent difference: \(String(format: "%.2f", percentDifference * 100))%")
+            
+            XCTAssertLessThanOrEqual(percentDifference, maxPercentRegression, "UT TDS performance regression exceeds allowed threshold")
+        }
+        
+        // Perform one last run inside measure block for XCTest metrics
+        measure {
+            _ = try? performSingleRun(tds: utTDS, run: numberOfRuns + 1, numberOfIterations: 1, name: "")
+        }
+    }
+        
+    func runPerformanceTest(tds: TrackerData, name: String) throws -> TimeInterval {
+        var allAverages: [TimeInterval] = []
+        
+        // Perform multiple runs outside of the measure block
+        for run in 1...numberOfRuns {
+            let average = try performSingleRun(tds: tds, run: run, numberOfIterations: numberOfIterationsPerRun, name: name)
+            allAverages.append(average)
+        }
+        
+        // Calculate and print the final average
+        let finalAverage = calculateFinalAverage(allAverages)
+        print("Final average for \(name) (30-70 percentile): \(finalAverage)")
+        
+        return finalAverage
+    }
+    
+    func performSingleRun(tds: TrackerData, run: Int, numberOfIterations: Int, name: String) throws -> TimeInterval {
```

Comment:
This measures both JSON encoding, content rule list store preparation and compilation - is this intended, or shall we focus only on compilation?

Reply:
Measuring is between CACurrentMediaTime() calls, so it should measure `store.compileContentRuleList`. 
I guess I'm missing your point.

Reply:
I think the point is that the `measure` block in the unit test is running all of it. It kind of defeats the purpose of using `measure` if you're doing your own measurement internally, might as well use a for loop (measure executes a whole bunch of times) so either:

1/ Refactor so that measure really is only measuring the compilation
or
2/ Just change measure to to a for-loop

But I think using measure is more useful as you can get baselines in Xcode.

Reply:
Yup, let's remove it. :)

Reply:
Done.